TCC.Lib/Benchmark/BenchmarkHelper.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using TCC.Lib.Blocks;
+using TCC.Lib.Options;
+
+namespace TCC.Lib.Benchmark
+{
+    public static class BenchmarkHelper
+    {
+        private static IEnumerable<BenchmarIteration> GenerateBenchmarkIteration(this BenchmarkOption benchmarkOption)
+        {
+            var algos = new[] {
+                CompressionAlgo.Lz4,
+                CompressionAlgo.Brotli, 
+                CompressionAlgo.Zstd
+            };
+            var withEncryption = new[] { false, true };
+
+            foreach (CompressionAlgo algo in algos)
+            {
+                int ratioMax;
+                switch (algo)
+                {
+                    case CompressionAlgo.Lz4:
+                        ratioMax = 9;
+                        break;
+                    case CompressionAlgo.Brotli:
+                        ratioMax = 9;
+                        break;
+                    case CompressionAlgo.Zstd:
+                        ratioMax = 19;
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use a constructor overloads that allows a more meaningful exception message to be provided. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3928)